tests/Scene/TestSelection.py

```diff
@@ -75,4 +79,60 @@ def test_applyOperation(self):
         # Node 1 got moved twice
         assert node_1.getPosition() == Vector(20, 40, 60)
         # And node 2 only once
-        assert node_2.getPosition() == Vector(10, 20, 30)
\ No newline at end of file
+        assert node_2.getPosition() == Vector(10, 20, 30)
+
+    def test_faceSelectMode(self):
+        Selection.selectedFaceChanged = MagicMock()
+
+        Selection.setFaceSelectMode(True)
+        assert Selection.getFaceSelectMode()
+
+        Selection.setFaceSelectMode(True)
+        Selection.setFaceSelectMode(False)
+        assert not Selection.getFaceSelectMode()
+        assert Selection.selectedFaceChanged.emit.call_count == 2
```

Comment:
Great that you test the exact call count. I sometimes feel that performance is lost by emitting duplicate signals

Reply:
Maybe we should come up with a more general strategy to mitigate that. Completely unsure _how_ though :-)